component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java

```diff
@@ -606,6 +606,16 @@ public void disconnect() {
             clientConnector = null;
             log.info("Server connector for url " + publisherURL + " disconnected.");
         }
+
+        if (httpConnectorFactory != null) {
+            try {
+                httpConnectorFactory.shutdown();
+                httpConnectorFactory = null;
+            } catch (InterruptedException e) {
+                log.info("Failed to shutdown the http connection factory while shutting down the siddhi app " +
+                        siddhiAppContext);
```

Comment:
`siddhiAppContext` should be `siddhiAppContext.getName()`

Reply:
Fixed in 2490e34